src/components/table/helpers/mixin-tbody-row.js

```diff
@@ -276,9 +285,37 @@ export default {
         }
 
         // Render the details slot in a TD
-        const $details = h(BTd, { props: { colspan: fields.length }, class: this.detailsTdClass }, [
-          this.normalizeSlot(detailsSlotName, detailsScope)
-        ])
+        var $details = !detailsScope.fields.some(e => e.details)
+          ? h(
+              BTd,
+              {
+                // Details row
+                props: {
+                  colspan: fields.length
+                }
+              },
+              [this.normalizeSlot(detailsSlotName, detailsScope)]
+            ) // Detailed slots
+          : detailsScope.fields.map((e, idx) =>
+              h(
+                BTd,
+                {
+                  attrs: {
+                    ...(e.details && {
+                      id: this.safeId('_col_details_'.concat(rowIndex, '_').concat(idx, '_'))
+                    })
+                  }
+                },
+                e.details
+                  ? [
+                      this.normalizeSlot(
+                        e.details === true ? detailsSlotName : e.details,
```

Comment:
This assumes `e.details` is the slot name, which it shouldn't it should be a boolean value.

The slot name should be `row-details(fieldKey)` where `fieldKey` is the name of the column.

Reply:
Javascript permissiveness allowed this funky thing. I undestand how it can be confusing. 

e.details was not intented to be strictly a boolean, it supposed to contain the slot name for the detail-td (where true should lead to the default slot "row-details" activating the showdetails row).

Following my inital logic, renamming "e.details" to "e.fieldDetailsSlotName" should describe better this intention, and the boolean testing should be replaced by a length or === "" comparaison.

Reply:
From my point of view, it could be fine to complement the rowdetails detection logic.

In the actual situation, row-details is only appenned when the v-slot:row-details is founded in the template (?).

It could be usefull to add an OR condition on it : 

If a `v-slot:row-details` is founded OR a `fieldDetailsSlotName` is founded in the fields : 

So the user would not be forced to name one of the details slot "row-details".

Reply:
(I did the changes and commited)